crates/router/src/connector/expresscheckout/transformers.rs

```diff
@@ -0,0 +1,326 @@
+use std::collections::HashMap;
+
+use masking::Secret;
+use serde::{Deserialize, Deserializer, Serialize};
+
+use crate::{
+    connector::utils::{self},
+    core::errors,
+    services::{self, api::request::Method},
+    types::{self, api, storage::enums},
+};
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+pub struct ExpresscheckoutPaymentsRequest {
+    #[serde(rename = "order.order_id")]
+    order_id: String,
+    #[serde(rename = "order.amount")]
+    amount: String,
+    #[serde(rename = "order.return_url")]
+    return_url: String,
+    #[serde(rename = "order.currency")]
+    currency: String,
+    #[serde(rename = "order.gateway_id")]
+    gateway_id: u8,
+    #[serde(flatten)]
+    gateway_reference_metadata: HashMap<String, String>,
+    merchant_id: String,
+    payment_method_type: PaymentMethodType,
+    card_number: Secret<String, common_utils::pii::CardNumber>,
+    card_exp_month: Secret<String>,
+    card_exp_year: Secret<String>,
+    name_on_card: Secret<String>,
+    card_security_code: Secret<String>,
+    format: String,
+    save_to_locker: bool,
+}
+
+#[derive(Serialize, Debug, Default, Eq, PartialEq)]
+enum PaymentMethodType {
+    #[default]
+    Card,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct GatewayMetadata {
+    gateway_id: u8,
+    gateway_reference_key: String,
+    gateway_reference_value: String,
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for ExpresscheckoutPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let metadata = item.connector_meta_data.clone().ok_or(
+            errors::ConnectorError::MissingRequiredField {
+                field_name: ("metadata"),
+            },
+        )?;
+        let gateway_metadata: GatewayMetadata = serde_json::from_value(metadata)
+            .map_err(|_| errors::ConnectorError::NoConnectorMetaData)?;
```

Comment:
This should be a parsing error or request encoding failed since the missing required field error has already been thrown above.